samples/hssi/hssi_100g_cmd.h

```diff
@@ -113,6 +141,212 @@ class hssi_100g_cmd : public hssi_cmd
+    return os;
+  }
+
+  std::ostream & print_monitor_data(std::ostream &os, perf_data *data, int port) const
+  {
+    os << std::dec
+    << std::left
+    << "| "
+    << std::setw(5)
+    << port << " | " << std::setw(15)
+    << data->tx_count << " | " << std::setw(15)
+    << data->rx_count << " | " << std::setw(15)
+    << data->rx_good_packet_count<< " | " << std::setw(10)
+    << data->rx_pkt_sec << " | " << std::setw(3)
+    << data->bw << "Gbps" << std::setw(5) << " | "
+    << "\r";
+    return os;
+  }
+
+  void select_port(int port, ctrl_config config_data, hssi_afu *hafu) const
+  {
+    /* Selects the port before performing read/write to traffic controller reg space */
+    hafu->write64(TRAFFIC_CTRL_PORT_SEL, port);
+    write_ctrl_config(hafu, config_data);
+    write_csr_addr(hafu, mac_bits_for(src_addr_), mac_bits_for(dest_addr_));
+  }
+
+  void capture_perf(perf_data *old_perf_data, hssi_afu *hafu) const
+  {
+    memset(old_perf_data, 0, sizeof(perf_data));
+    if (stall_enable(hafu))
+    {
+      read_performance(old_perf_data, hafu);
+      stall_disable(hafu);
+    }
+  }
+
+  std::ostream & run_monitor(std::ostream &os, perf_data *old_perf_data, int port, hssi_afu *hafu)
+  {
+    perf_data *cur_perf_data = (perf_data*)malloc(sizeof(perf_data));
+    perf_data *new_perf_data = (perf_data*)malloc(sizeof(perf_data));
+
+    memset(cur_perf_data, 0, sizeof(perf_data));
+
+    capture_perf(new_perf_data, hafu);
+    calc_performance(old_perf_data, new_perf_data, cur_perf_data, start_size_);
+
+    print_monitor_data(os, cur_perf_data, port);
+    memcpy(old_perf_data, new_perf_data, sizeof(perf_data));
+
+    return os;
+  }
+
+  char get_user_input(char key) const
+  {
+    /* Waits for user input for 1 second */
+    key = '\0';      // initialized to null
+    std::thread t1([&](){
+      std::cin>>key;
+    });
+    t1.detach();
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+    return key;
+  }
+
+  void quit_monitor(uint8_t size, std::string move, hssi_afu *hafu) const
+  {
+    /* Quit the monitor*/
+    move_cursor(size, move);
+    hafu->mbox_write(CSR_CTRL0, DATA_CNF_PKT_NUM);
+    hafu->mbox_write(CSR_STATS_CTRL, ZERO);
+  }
+
+  void reset_monitor(std::vector<int> port_, std::string move, ctrl_config config_data, hssi_afu *hafu) const
+  {
+    /* Reset the monitor ports traffic data*/
+    uint8_t size = (uint8_t)port_.size();
+    for(uint8_t port=0; port < size; port++)
+      {
+        select_port(port_[port], config_data, hafu);
+        hafu->mbox_write(CSR_STATS_CTRL, ONE);
+        hafu->mbox_write(CSR_STATS_CTRL, ZERO);
+      }
+     move_cursor(1, move);
+  }
+
+  void move_cursor(uint8_t size, std::string move) const
+  {
+    /* Moves the cursor position (up/down) based on port size*/
+    char cursor[20] = "\x1b[";
+    char size_ptr[5];
+    sprintf(size_ptr, "%d", size);
```

Comment:
Can we use std::ostringstream instead to keep it in the domain of C++?

Reply:
Yes we can , we will make this change